### ISSUE TITLE
Problem with astropy 4.2 numpy requirements

### ISSUE THREAD

**Issue by mwcraig** [ContinuumIO/anaconda-issues#12269]
TL;DR: The astropy 4.2 recipe is conda-forge was broken (wrong numpy version in the recipe) and has been fixed. The defaults channel has the incorrect version of  the numpy dependencies. Working conda-forge recipe is at https://github.com/conda-forge/astropy-feedstock

### Actual Behavior

Some supported combinations of astropy and numpy do not install because of problems with the astropy version requirements for numpy.

### Expected Behavior

This should work, since the packages numpy 1.18 and astropy 4.2 are on the defaults channel, and astropy 4.2 supports numpy >=1.17. 

### Steps to Reproduce

```
conda create -n astropy-fail --override-channels -c defaults astropy=4.2 numpy=1.18 python=3.8
```

produces this output:

```
conda create -n astropy-fail --override-channels -c defaults astropy=4.2 numpy=1.18 python=3.8
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: -
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package python conflicts for:
astropy=4.2 -> python[version='>=3.7,<3.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0']
python=3.8
astropy=4.2 -> numpy[version='>=1.16.6,<2.0a0'] -> python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0']

Package numpy conflicts for:
astropy=4.2 -> numpy[version='>=1.16|>=1.16.6,<2.0a0|>=1.19.2,<2.0a0']
numpy=1.18
astropy=4.2 -> pyerfa -> numpy[version='>=1.17.0,<2.0a0']
```

##### Anaconda or Miniconda version:

I'm not sure, tbh. I installed it a long time ago...

##### `conda info`

<details>

**NOTE:** I overrode the channels in the command above to make sure the solver was only using defaults.

```
     active environment : base
    active env location : /Users/mattcraig/miniconda3
            shell level : 2
       user config file : /Users/mattcraig/.condarc
 populated config files : /Users/mattcraig/.condarc
          conda version : 4.9.2
    conda-build version : 3.18.11
         python version : 3.7.3.final.0
       virtual packages : __osx=10.15.7=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /Users/mattcraig/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/vpython/osx-64
                          https://conda.anaconda.org/vpython/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /Users/mattcraig/miniconda3/pkgs
                          /Users/mattcraig/.conda/pkgs
       envs directories : /Users/mattcraig/miniconda3/envs
                          /Users/mattcraig/.conda/envs
               platform : osx-64
             user-agent : conda/4.9.2 requests/2.24.0 CPython/3.7.3 Darwin/19.6.0 OSX/10.15.7
                UID:GID : 502:20
             netrc file : None
           offline mode : False

```

</details>

##### `conda list --show-channel-urls`

<details>

```
# packages in environment at /Users/mattcraig/miniconda3:
#
# Name                    Version                   Build  Channel
anaconda-client           1.7.2                    py37_0    defaults
atomicwrites              1.4.0                      py_0    defaults
attrs                     20.3.0             pyhd3eb1b0_0    defaults
beautifulsoup4            4.9.3              pyhb0f4dca_0    defaults
bleach                    3.2.1                      py_0    defaults
brotlipy                  0.7.0           py37h9ed2024_1003    defaults
bzip2                     1.0.8                h1de35cc_0    defaults
ca-certificates           2021.1.19            hecd8cb5_0    defaults
certifi                   2020.12.5        py37hecd8cb5_0    defaults
cffi                      1.14.0           py37hb5b8e2f_0    defaults
chardet                   3.0.4           py37hecd8cb5_1003    defaults
clyent                    1.2.2                    py37_1    defaults
cmarkgfm                  0.4.2            py37h1de35cc_0    defaults
conda                     4.9.2            py37hecd8cb5_0    defaults
conda-build               3.18.11                  py37_0    defaults
conda-package-handling    1.7.2            py37h22f3db7_0    defaults
cryptography              3.2.1            py37hbcfaee0_1    defaults
cython                    0.29.13          py37h0a44026_0    defaults
docutils                  0.16                     py37_1    defaults
filelock                  3.0.12                     py_0    defaults
future                    0.18.2                   py37_1    defaults
glob2                     0.7                        py_0    defaults
icu                       58.2                 h0a44026_3    defaults
idna                      2.10                       py_0    defaults
importlib-metadata        2.0.0                      py_1    defaults
importlib_metadata        2.0.0                         1    defaults
ipython_genutils          0.2.0                    py37_0    defaults
jinja2                    2.10.3                     py_0    defaults
jsonschema                3.2.0                      py_2    defaults
jupyter_core              4.6.3                    py37_0    defaults
krb5                      1.18.2               h75d18d8_0    defaults
libarchive                3.5.1                h078526d_0    conda-forge
libcurl                   7.71.1               h8a08a2b_1    defaults
libcxx                    11.0.1               habf9029_0    conda-forge
libedit                   3.1.20191231         h1de35cc_1    defaults
libffi                    3.2.1             h0a44026_1007    defaults
libiconv                  1.16                 h1de35cc_0    defaults
liblief                   0.10.1               h0a44026_0    defaults
libsolv                   0.7.17               hfd3ada9_0    conda-forge
libssh2                   1.9.0                ha12b0ac_1    defaults
libxml2                   2.9.10               h7cdb67c_3    defaults
lz4-c                     1.9.2                h79c402e_3    defaults
lzo                       2.10                 haf1e3a3_2    defaults
mamba                     0.7.12           py37h85ac496_0    conda-forge
markupsafe                1.1.1            py37h1de35cc_0    defaults
more-itertools            8.6.0              pyhd3eb1b0_0    defaults
nbformat                  5.0.8                      py_0    defaults
ncurses                   6.2                  h0a44026_1    defaults
nodejs                    10.13.0              h0a44026_0    defaults
openssl                   1.1.1i               h9ed2024_0    defaults
packaging                 20.4                       py_0    defaults
pip                       20.2.4           py37hecd8cb5_0    defaults
pkginfo                   1.6.1            py37hecd8cb5_0    defaults
pluggy                    0.13.1                   py37_0    defaults
psutil                    5.7.2            py37haf1e3a3_0    defaults
py                        1.9.0                      py_0    defaults
py-lief                   0.10.1           py37haf313ee_0    defaults
pycosat                   0.6.3            py37h9ed2024_0    defaults
pycparser                 2.20                       py_2    defaults
pygments                  2.7.2              pyhd3eb1b0_0    defaults
pyopenssl                 19.1.0             pyhd3eb1b0_1    defaults
pyparsing                 2.4.7                      py_0    defaults
pyrsistent                0.17.3           py37haf1e3a3_0    defaults
pysocks                   1.7.1            py37hecd8cb5_0    defaults
pytest                    5.2.1                    py37_0    defaults
python                    3.7.3                h93065d6_1    conda-forge
python-dateutil           2.8.1                      py_0    defaults
python-libarchive-c       2.9                        py_0    defaults
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2020.1                     py_0    defaults
pyyaml                    5.3.1            py37haf1e3a3_1    defaults
readline                  8.0                  h1de35cc_0    defaults
readme_renderer           24.0                     py37_0    defaults
reproc                    14.2.1               hbcb3906_0    conda-forge
reproc-cpp                14.2.1               h2e338ed_0    conda-forge
requests                  2.24.0                     py_0    defaults
requests-toolbelt         0.9.1                      py_0    defaults
ripgrep                   12.1.1                        0    defaults
ruamel_yaml               0.15.87          py37haf1e3a3_1    defaults
setuptools                50.3.1           py37hecd8cb5_1    defaults
six                       1.15.0           py37hecd8cb5_0    defaults
soupsieve                 2.0.1                      py_0    defaults
sqlite                    3.33.0               hffcf06c_0    defaults
tk                        8.6.10               hb0a8c7a_0    defaults
tqdm                      4.51.0             pyhd3eb1b0_0    defaults
traitlets                 5.0.5                      py_0    defaults
twine                     2.0.0                      py_0    defaults
urllib3                   1.25.11                    py_0    defaults
wcwidth                   0.2.5                      py_0    defaults
webencodings              0.5.1                    py37_1    defaults
wheel                     0.35.1             pyhd3eb1b0_0    defaults
xz                        5.2.5                h1de35cc_0    defaults
yaml                      0.2.5                haf1e3a3_0    defaults
you-get                   0.4.1432                 py37_0    local
zipp                      3.4.0              pyhd3eb1b0_0    defaults
zlib                      1.2.11               h1de35cc_3    defaults
zstd                      1.4.5                h41d2c2f_0    defaults

```

</details>

**Comment by chenghlee** [comment]
Reopening; also a dependencies-specification error in the Python 3.8 `pyerfa` package.

**Comment by mwcraig** [comment]
Thanks for catching pyerfa and for the quick fixes!

**Comment by csoja** [comment]
@chenghlee Is this fixed now? Can it be closed?
